enterprise/mqtt/src/main/java/io/crate/mqtt/operations/MqttIngestService.java

```diff
@@ -168,24 +180,66 @@ public void doInsert(String clientId, MqttPublishMessage msg, BiConsumer<Object,
         List<Object> argsAsList = Arrays.asList(args);
 
         boolean messageMatchedRule = false;
+        boolean callbackNotified = false;
         SQLOperations.Session session = sqlOperations.createSession(Schemas.DOC_SCHEMA_NAME, crateUser, Option.NONE, 1);
+        List<CompletableFuture> insertOperationsFuture = new ArrayList<>(predicateAndIngestRules.size());
```

Comment:
Please try to use generics, even if it's a wildcard like `List<CompletableFuture<?>>` (this would enable us to get rid of the warning further below in ` for (CompletableFuture result : insertOperationsFuture) {`